coreUtilities/src/main/java/org/opensearch/migrations/transformation/rules/IndexMappingTypeRemoval.java

```diff
@@ -0,0 +1,54 @@
+package org.opensearch.migrations.transformation.rules;
+
+import java.util.Map.Entry;
+
+import org.opensearch.migrations.transformation.CanApplyResult;
+import org.opensearch.migrations.transformation.CanApplyResult.Unsupported;
+import org.opensearch.migrations.transformation.TransformationRule;
+import org.opensearch.migrations.transformation.entity.Index;
+
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+/**
+ * Supports transformation of the Index Mapping types that were changed from mutliple types to a single type between ES 6 to 7
+ */
+public class IndexMappingTypeRemoval implements TransformationRule<Index> {
+
+    @Override
+    public CanApplyResult canApply(final Index index) {
+        final var mappingNode = index.raw().get("mappings");
```

Comment:
Won't you need the path (or maybe even headers) to make some determinations here in the general case too?  How will this fit in with the IHttpMessage defined in the existing transformerPlugins?

This seems like it will work great for the Metadata Migration case, but having to create a specific Index type for replays could be an assumption that makes the code more brittle and less performant.